enterprise/frontend/src/embedding-sdk/components/public/CollectionBrowser/CollectionBrowser.tsx

```diff
@@ -0,0 +1,65 @@
+import { useState } from "react";
+
+import { withPublicComponentWrapper } from "embedding-sdk/components/private/PublicComponentWrapper";
+import {
+  ALL_MODELS,
+  COLLECTION_PAGE_SIZE,
+} from "metabase/collections/components/CollectionContent";
+import { CollectionItemsTable } from "metabase/collections/components/CollectionContent/CollectionItemsTable";
+import CollectionBreadcrumbs from "metabase/nav/containers/CollectionBreadcrumbs/CollectionBreadcrumbs";
+import type {
+  CollectionEssentials,
+  CollectionId,
+  CollectionItem,
+  CollectionItemModel,
+} from "metabase-types/api";
+
+type CollectionBrowserProps = {
+  collectionId: CollectionId;
+  onClick?: (item: CollectionItem) => void;
+  pageSize?: number;
+  visibleCollectionTypes?: CollectionItemModel[];
+};
+
+export const CollectionBrowser = withPublicComponentWrapper(
+  ({
+    collectionId,
+    onClick,
+    pageSize = COLLECTION_PAGE_SIZE,
+    visibleCollectionTypes = ALL_MODELS,
+  }: CollectionBrowserProps) => {
+    const [currentCollectionId, setCurrentCollectionId] =
+      useState(collectionId);
+
+    const onClickItem = (item: CollectionItem) => {
+      if (onClick) {
+        onClick(item);
+      }
+
+      if (item.model === "collection") {
+        setCurrentCollectionId(item.id);
+      }
+    };
+
+    const onClickBreadcrumbItem = (item: CollectionEssentials) => {
+      setCurrentCollectionId(item.id);
+    };
+
+    return (
+      <>
+        <CollectionBreadcrumbs
```

Comment:
I'm using the breadcrumbs here to allow users to navigate. It takes in the given collection ID, and then the breadcrumbs use that ID as the base entity to base the rest of the breadcrumbs off of. This allows users to limit navigation only to entities under the original `collectionId`